sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/WebExceptionRetryPolicy.java

```diff
@@ -5,65 +5,85 @@
 
 import com.azure.cosmos.implementation.apachecommons.lang.time.StopWatch;
 import com.azure.cosmos.implementation.directconnectivity.WebExceptionUtility;
+import com.azure.cosmos.implementation.http.HttpTimeoutPolicy;
+import com.azure.cosmos.implementation.http.HttpTimeoutPolicyDefault;
+import com.azure.cosmos.implementation.http.ResponseTimeoutAndDelays;
+import io.netty.handler.codec.http.HttpMethod;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import reactor.core.publisher.Mono;
 
 import java.time.Duration;
 import java.util.concurrent.TimeUnit;
 
-public class WebExceptionRetryPolicy implements IRetryPolicy {
+public class WebExceptionRetryPolicy extends DocumentClientRetryPolicy {
     private final static Logger logger = LoggerFactory.getLogger(WebExceptionRetryPolicy.class);
 
-    // total wait time in seconds to retry. should be max of primary reconfigrations/replication wait duration etc
-    private final static int waitTimeInSeconds = 30;
-    private final static int initialBackoffSeconds = 1;
-    private final static int backoffMultiplier = 2;
-
     private StopWatch durationTimer = new StopWatch();
-    private int attemptCount = 1;
-    // Don't penalise first retry with delay.
-    private int currentBackoffSeconds = WebExceptionRetryPolicy.initialBackoffSeconds;
+    private int backoffSecondsTimeout;
     private RetryContext retryContext;
+    private RxDocumentServiceRequest request;
+    private HttpTimeoutPolicy timeoutPolicy;
+    private HttpMethod httpMethod;
+    private int retryCountTimeout = 0;
 
     public WebExceptionRetryPolicy() {
-        durationTimer.start();
+        this(null);
     }
 
     public WebExceptionRetryPolicy(RetryContext retryContext) {
         durationTimer.start();
         this.retryContext = retryContext;
+        this.timeoutPolicy = HttpTimeoutPolicyDefault.INSTANCE;
     }
 
     @Override
     public Mono<ShouldRetryResult> shouldRetry(Exception exception) {
-        Duration backoffTime = Duration.ofSeconds(0);
-
+        boolean isOutOfRetries = isOutOfRetries();
+        if (isOutOfRetries) {
+            this.durationTimer.stop();
+            return Mono.just(ShouldRetryResult.noRetry());
+        }
+        if (!timeoutPolicy.isSafeToRetry(httpMethod) && !WebExceptionUtility.isWebExceptionRetriable(exception)) {
```

Comment:
which condition will be covered here but not covered by`   if (!WebExceptionUtility.isWebExceptionRetriable(exception)) {` ?